server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -77,6 +77,16 @@ public final long getUsageCount() {
 
     @Override
     public final void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        // check if the query has any parameters that are not in the supported set (if declared)
+        var supported = supportedQueryParameters();
+        if (supported != null) {
+            Set<String> unsupported = new HashSet<>(request.params().keySet());
```

Comment:
Is it worth avoiding the allocation on the happy path?

Reply:
Ah yes, good idea